lms/templates/main.html

```diff
@@ -193,7 +193,7 @@
     <div class="marketing-hero"><%block name="marketing_hero"></%block></div>
 
     <div class="content-wrapper main-container" id="content" dir="${static.dir_rtl()}">
-      ${self.body()}
+      ${next.body()}
```

Comment:
what does this do?

Reply:
See https://docs.makotemplates.org/en/latest/inheritance.html#using-the-next-namespace-to-produce-content-wrapping.

In summary, the `self` namespace refers to the body of the "most derived" template in the inheritance tree. So right now, [`main.html`](https://github.com/edx/edx-platform/blob/6ed1ae05cbc972620f42e67d9d0642d8ae6e7b0f/lms/templates/main.html) calls [`full_fbe.html`](https://github.com/edx/edx-platform/blob/6ed1ae05cbc972620f42e67d9d0642d8ae6e7b0f/lms/templates/course_modes/track_selection_types/full_fbe.html), skipping [`track_selection.html`](https://github.com/edx/edx-platform/blob/6ed1ae05cbc972620f42e67d9d0642d8ae6e7b0f/lms/templates/course_modes/track_selection.html).

Instead, I want `main.html` to call the body of `track_selection.html`, and then for `track_selection.html` to call the blocks in `full_fbe.html`.

This requires `next.body()` to be called in `main.html` instead of `self.body()`.

I did a check to see if this would break anything in edx-platform, and my opinion is that it won't. For my research notes, please see [this page in Confluence](https://openedx.atlassian.net/wiki/spaces/~931919476/pages/3147530374/REV-2325#Mako-Research).

Reply:
Interesting- thanks!